src/org/traccar/protocol/TeltonikaProtocolDecoder.java

```diff
@@ -134,15 +134,92 @@ private void decodeParameter(Position position, int id, ChannelBuffer buf, int l
             case 78:
                 position.set(Position.KEY_DRIVER_UNIQUE_ID, String.valueOf(readValue(buf, length, false)));
                 break;
+            case 80:
+                position.set("workMode", readValue(buf, length, false));
+                break;
+            case 179:
+                position.set(Position.PREFIX_OUT + 1, readValue(buf, length, false) == 1);
+                break;
+            case 180:
+                position.set(Position.PREFIX_OUT + 2, readValue(buf, length, false) == 1);
+                break;
+            case 181:
+                position.set(Position.KEY_PDOP, readValue(buf, length, false) * 0.1);
+                break;
             case 182:
                 position.set(Position.KEY_HDOP, readValue(buf, length, false) * 0.1);
                 break;
+            case 239:
+                position.set(Position.KEY_IGNITION, readValue(buf, length, false) == 1);
+                break;
+            case 240:
+                position.set(Position.KEY_MOTION, readValue(buf, length, false) == 1);
+                break;
+            case 241:
+                position.set(Position.KEY_OPERATOR, readValue(buf, length, false));
+                break;
+            default:
+                position.set(Position.PREFIX_IO + id, readValue(buf, length, false));
+                break;
+        }
+    }
+
+    private void decodeGh3000Parameter(Position position, int id, ChannelBuffer buf, int length) {
+        switch (id) {
+            case 1:
+                position.set(Position.KEY_BATTERY_LEVEL, readValue(buf, length, false));
+                break;
+            case 2:
+                position.set("usbConnected", readValue(buf, length, false) == 1);
+                break;
+            case 5:
+                position.set("uptime", readValue(buf, length, false));
+                break;
+            case 20:
+                position.set(Position.KEY_HDOP, readValue(buf, length, false) * 0.1);
+                break;
+            case 21:
+                position.set(Position.KEY_VDOP, readValue(buf, length, false) * 0.1);
+                break;
+            case 22:
+                position.set(Position.KEY_PDOP, readValue(buf, length, false) * 0.1);
+                break;
+            case 67:
+                position.set(Position.KEY_BATTERY, readValue(buf, length, false) * 0.001);
+                break;
+            case 221:
+                position.set("button", readValue(buf, length, false));
+                break;
+            case 222:
+                if (readValue(buf, length, false) == 1) {
+                    position.set(Position.KEY_ALARM, Position.ALARM_SOS);
+                }
+                break;
+            case 240:
+                position.set(Position.KEY_MOTION, readValue(buf, length, false) == 1);
+                break;
+            case 244:
+                position.set("roaming", readValue(buf, length, false) == 1);
```

Comment:
I believe we can have a dedicated key for this. I recall there are some other protocol that report this information.

Reply:
Yes, it worth to be dedicated key. But only one place I've found with "roam/roaming" words is https://github.com/tananaev/traccar/blob/master/src/org/traccar/protocol/Gl200BinaryProtocolDecoder.java#L267